src/frequenz/sdk/actor/_power_managing/_base_classes.py

```diff
@@ -72,6 +73,80 @@ class Report:
     This is `None` if no power distribution has been performed yet.
     """
 
+    @property
+    def bounds(self) -> timeseries.Bounds[Power] | None:
+        """The bounds for the batteries.
+
+        These bounds are adjusted to any restrictions placed by actors with higher
+        priorities.
+
+        There might be exclusion zones within these bounds. If necessary, the
+        [`adjust_to_bounds`][frequenz.sdk.timeseries.battery_pool.Report.adjust_to_bounds]
+        method may be used to check if a desired power value fits the bounds, or to get
+        the closest possible power values that do fit the bounds.
+        """
+        return self._inclusion_bounds
+
+    def adjust_to_bounds(self, power: Power) -> tuple[Power | None, Power | None]:
```

Comment:
Still need to figure out what the right return type is.  They are not bounds, but the closest available power values that can be proposed to the battery pool.

That said,  this whole method can be safely ignored and people can just rely on the PowerManager, so there's no need to make this more clear.

There's also an example below in this method's docstring, that shows how it can be used with a `match` statement.  Maybe that's all that's needed.

Reply:
Why publishing it if you don't have to use it then?

If you want to keep it, about the type, maybe `PowerRange`?

![PowerRangers](https://media.nbcboston.com/2023/01/GettyImages-819239982.jpg?quality=85&strip=all&crop=0px%2C161px%2C2752px%2C1548px&resize=1200%2C675)

Reply:
Except it is not a `Range` either,  they are the two best possible options.  

Most users shouldn't have to worry about this method, but it  might be useful in one specific situation:  If an actor sets bounds that fall entirely within the exclusion bounds range, they might want to be able to choose which side of the exclusion bounds they want to go to.

Reply:
I'm open to removing it and adding it if there seems to be a need for such flexibility among users.  Especially in cases where there's only a single actor running, they'll be able to get more control by using this, if they want.

Reply:
I guess this is one of the problems with trying to stabilize an API that has never been used in production before.

Maybe we just prefix with `_` and call it experimental for now?  Or maybe prefix with `experimental_`, so automatic documentation works?

Reply:
Didn't do anything, leave it as it is.  The function returns a pair of powers, which don't have two separate names, just two options, so I think a tuple makes sense.

Reply:
I'm all for leaving it with `_`, so if someone needs it, they can use it already, and we don't need to remove it now and add it later. If people really start to use it and it works, we make it public. I really like this approach as "the best of both worlds".

Reply:
is there anyway to override the doc generator to show docs for this function even if it is `_`?

Reply:
No, but that's the point, right? That it is hidden, but if someone asks for the functionality we can say "use this"?

Reply:
This method's documentation provides a lot of context about the behaviour of the bounds, so I'd like it to be visible,  but I'll add a line saying that the method is experimental.

and it looks like it is possible:  https://mkdocstrings.github.io/python/usage/configuration/members/

I will test.

Reply:
In the API that's not really possible, as all of that is auto-generated and we don't have a way to override how we render a particular module or symbol. You could pull that documentation into some other doc in `docs/`. What I don't like about making it public is that autocompletion might suggest it and users might not look at the docs in detail, and after you are using it, there no indication that you are using something experimental.

With `_` you don't get auto-completion and the code is marked with `# pylint: disable=private-access` (or whatever is called), which clearly shows something is fishy, and it is also clear that private interfaces are not stable.

Can you put that documentation somewhere else? Maybe in the class, saying "There is an experimental `_adjust_to_bounds()` that does this and that"?

Reply:
> What I don't like about making it public is that autocompletion might suggest it and users might not look at the docs in detail, and after you are using it, there no indication that you are using something experimental.

I mean, this whole thing is one big experimental feature, not just this function.  We're basically changing all of these interfaces and trying to release them before we've had a chance to test them.

Reply:
Fair enough!